install-and-configure/install/multi-cluster/federated-etl/aggregator.md

```diff
@@ -0,0 +1,143 @@
+    # governs parallelism of derivation step
+    # more threads speeds derivation, but requires significantly more 
+    # disk space for WAL storage
+    # default: 6
+    NUM_DERIVER_THREADS: 6
+    # log level
+    # default: info
+    LOG_LEVEL: debug
+  aggregatorDbStorage:
+    # governs storage size of aggregator DB storage
+    # !!NOTE!! disk performance is _critically important_ to aggregator performance
+    # ensure disk is specd high enough, and check for bottlenecks
+    # default: 128Gi
+    storageRequest: 128Gi
+  # tracing, starts embedded jaeger pod
+  # port forward to 16686 on the pod to access jaeger UI 
+  # provides instrumented traces used for troubleshooting performance issues
+  jaeger:
+    enabled: true
+    image: jaegertracing/all-in-one
+    imageVersion: latest
+federatedETL:
+  federatedCluster: true
+kubecostModel:
+  containerStatsEnabled: true
+  federatedStorageConfigSecret: federated-store
+kubecostProductConfigs:
+  clusterName: YOUR_CLUSTER_NAME
+  cloudIntegrationSecret: cloud-integration
+  productKey:
+    enabled: true
+    key: YOUR_KEY
+prometheus:
+  server:
+    global:
+      external_labels:
+        cluster_id: YOUR_CLUSTER_NAME
+# when using managed identity/irsa, set the service account accordingly:
+serviceAccount:
+  create: false
+  name: kubecost-irsa-sa
+```
+
+There is no baseline for what is considered a larger deployment, which will be dependent on load times in your Kubecost environment.
+
+Once you’ve configured your *aggregator.yaml*, create secrets using the following commands:
+
+{% code overflow="wrap" %}
+```
+kubectl create secret generic cloud-integration -n kubecost --from-file=aggregator.yaml
+```
+{% endcode %}
+
+{% code overflow="wrap" %}
+```
+kubectl create secret generic federated-storage -n kubecost --from-file=aggregator.yaml
+```
+{% endcode %}
+
+Finally, upgrade your existing Kubecost installation. This command will install Kubecost if it does not already exist:
+
+{% hint style="warning" %}
+Upgrading your existing Kubecost using your configured *aggregator.yaml* file above will reset all existing Helm values configured in your *values.yaml*. If you wish to preserve any of those changes, append your *values.yaml* by adding the contents of your *aggregator.yaml* file into it, then replacing `aggregator.yaml` with `values.yaml` in the upgrade command below:
+{% endhint %}
+
+```
+helm upgrade --install "kubecost-primary" \
+--namespace kubecost-primary \
+--repo https://kubecost.github.io/cost-analyzer/ cost-analyzer \
+-f aggregator.yaml
+```
+
+### Validating Aggregator pod is running successfully
+
+When first enabled, the aggregator pod will ingest the last 90 days of ETL data from the federated-store. This may take several hours. Because the combined folder is ignored, the federator pod is not used here, but can still run if needed. You can run `kubectl get pods` and ensure the `aggregator` pod is running, but should still wait for all data to be ingested.

```

Comment:
> the aggregator pod will ingest the last 90 days of ETL data

By default I believe this is actually something like 3 years because of the quarterly default value. We may want to change those defaults @ameijer

Reply:
yeah 90 days is wrong, good catch. I kind of like leaving the 3 year idea though?? to showcase what this is capable of?

Reply:
Sure, maybe we cut it down for free but leave as-is for enterprise?

Reply:
any ideas as how to implement that? my thinking was the window restriction middleware would accomplish that anyway?

Reply:
Have a set of free-only default env vars (to reduce ingest) and then rely on middleware for query-time gating

Reply:
Ingest reduction being mainly for perf purposes